File: public/samples/DataFeeds/SVR/broadcaster.go
import "github.com/flashbots/mev-share-node/mevshare"

// ... create and sign your transaction
// Your transaction should send ETH to block.coinbase as the tip/bid
// Example: Include a call that does payable(block.coinbase).transfer(msg.value)
tx := types.NewTransaction(...)
signedTx, err := types.SignTx(tx, ...)
txBytes := signedTx.MarshalBinary()

bundle := []mevshare.MevBundleBody {
    {
        Hash: &event.EventHash,
    },
    {
        Tx:        &txBytes,
        CanRevert: false,
    },
}

params := mevshare.SendMevBundleArgs {
    Body:      bundle,
    Inclusion: mevshare.MevBundleInclusion{blockNumber, blockNumber+1},
}

byteParams, err := json.Marshal(params)
body := []byte(fmt.Sprintf(`{"jsonrpc":"2.0","method":"mev_sendBundle","params":["%s"], "id":1}`, string(byteParams)))
bodyReader := bytes.NewReader(body)
postReq, err := http.NewRequestWithContext(ctx, http.MethodPost, "https://relay.flashbots.net/", bodyReader)

hashedBody := crypto.Keccak256Hash(body).Hex()
signedMessage, err := crypto.Sign(accounts.TextHash([]byte(hashedBody)), privKey)
postReq.Header.Add("X-Flashbots-signature", address.String()+":"+hexutil.Encode(signedMessage))
resp, err := http.DefaultClient.Do(postReq)

File: public/samples/DataFeeds/SVR/broadcaster.ts
async function sendBackrunBundle(pendingTx: IPendingTransaction) {
  const currentBlock = await provider.getBlockNumber()
  const signedBackrunTx = await buildDummyBackrunTx(pendingTx)
  const bundleParams: BundleParams = {
    inclusion: {
      block: currentBlock + 1,
      maxBlock: currentBlock + TARGET_BUNDLE_BLOCK_DELAY,
    },
    body: [{ hash: pendingTx.hash }, { tx: signedBackrunTx, canRevert: false }],
  }

  const sendBundleResult = await mevShareClient.sendBundle(bundleParams)
  return sendBundleResult
}

async function buildDummyBackrunTx(pendingTx: IPendingTransaction) {
  // Your liquidation transaction should send ETH to block.coinbase as the tip/bid
  // Example: Include a call that does payable(block.coinbase).transfer(tipAmount)
  const backrunTx = { data: {}, maxFeePerGas: 22000, maxPriorityFeePerGas: 22000 }
  const signedBackrunTx = await executorWallet.signTransaction(backrunTx)
  return signedBackrunTx
}
